homeassistant/components/shelly/__init__.py

```diff
@@ -216,6 +218,10 @@ def _async_device_online(_: Any, update_type: BlockUpdateType) -> None:
             raise ConfigEntryNotReady(repr(err)) from err
         except InvalidAuthError as err:
             raise ConfigEntryAuthFailed(repr(err)) from err
+        except FirmwareUnsupported as err:
+            error = repr(err)
+            async_create_issue_unsupported_firmware(hass, entry)
+            raise ConfigEntryNotReady(error) from err
```

Comment:
```suggestion
            async_create_issue_unsupported_firmware(hass, entry)
            raise ConfigEntryNotReady from err
```

https://github.com/home-assistant/core/pull/109076/files#r1470309887